docs/contributing.md

```diff
@@ -0,0 +1,169 @@
+
+# Table of Contents
+
+1.  [Autorest.Typescript Dev Workflow](#org9754a77)
+    1.  [Repository](#org38012c9)
+    2.  [Set-up](#org422658b)
+    3.  [Autorest Core version](#orgfd864b4)
+    4.  [Debugging (VSCode)](#orgf583807)
+        1.  [Code Generation](#orgd6903ff)
+        2.  [Unit Tests](#org1fee673)
+        3.  [Integration Tests](#orgb632069)
+    5.  [Troubleshooting](#orgbd557ae)
+        1.  [Before investing more time investigating](#org70e1598)
+        2.  [I've made changes to Autorest.Typescript, however when re-generating the library my changes doesn't seem to have been picked up](#orge680821)
+
+
+<a id="org9754a77"></a>
+
+# Autorest.Typescript Dev Workflow
+
+
+<a id="org38012c9"></a>
+
+## Repository
+
+-   Autorest.Typescript is being developed in the Azure/Autorest.Typescript repository
+-   The version under development is known as v6 and the working branch is "v6", master contains the olver v5 version.
+-   In the new version the package name has changed to @autorest/typescript
+
+
+<a id="org422658b"></a>
+
+## Set-up
+
+-   Clone the repository[ Azure/Autorest.Typescript](https://github.com/Azure/autorest.typescript.git)
+-   Switch to branch v6
+    
+        git fetch upstream v6
+        git checkout v6
+
+-   Install Autorest Core
+    
+        npm install -g @autorest/autorest
```

Comment:
I see `"autorest": "^3.0.6187"` in the deps in `package.json`, is this still required? should `"autorest` be a peer dependency instead?

Reply:
This is recommended to be installed globally when you need to test code-generation outside of the Autorest.Typescript repo.

For example, say that the Azure Tables team report an issue with the generated output. You make the changes locally in Autorest.Typescript and want to test those changes in the azure-sdk-for-js repo. For this you will need to run `autorest --typescript --use=<Path to autorest.typescript repo>`